### PR TITLE
Readme and Tests

### PR DESCRIPTION
**Pull request by r4vi**
This fleshes out the readme and adds some basic tests but doesn't change the functionality of the proxy

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,4 +1,158 @@
 # Analytics Platform Cran Proxy
+
 [![Docker Repository on Quay](https://quay.io/repository/mojanalytics/cran-proxy/status "Docker Repository on Quay")](https://quay.io/repository/mojanalytics/cran-proxy)
 
+- [Description](#sec-1)
```
Comment (davidread) [inline README.md]: Numbers for ids are brittle, aren't they?

File: README.md [hunk 2]
```diff
@@ -1,4 +1,158 @@
...
+# Description<a id="sec-1"></a>
+
 A cran mirror that compiles binary packages for the rocker:verse environment.
+
+## Why<a id="sec-1-1"></a>
+
+CRAN servers usually provide binary versions of libraries for Windows and OSX
+but not for Linux. This is understandable because there's no such thing as
+*"Linux OS"*. Each distribution will have a set of C libraries, version of
+`libc` or a different implementation of `libc` altogether (`musl`).
```
Comment (davidread) [inline README.md]: just to be really clear perhaps add something like: ", meaning you'd have to compile hundreds of binaries for each R package, to cope with the myriad of linux options."

File: README.md [hunk 3]
```diff
@@ -1,4 +1,158 @@
...
+
+However, in our environment we've standardised on using the `rocker:verse`
+docker images images with a pinned version of R (3.4). Given we know the
+constraints it is possible to run a CRAN server that provides binary for
```
Comment (davidread) [inline README.md]: "binary" singular doesn't read right

File: README.md [hunk 4]
```diff
@@ -1,4 +1,158 @@
...
+
+## Administrators<a id="sec-4-2"></a>
+
+### Environment Variables<a id="sec-4-2-1"></a>
+
+| Name | Default | Description |
+|------------------------- |------------------------------ |-------------------------------------------------------------------------------------------------------- |
+| `DEBUG` | `FALSE` | If set to true the webserver will auto reload on changes, dump exceptions and set the log level to DEBUG |
+| `PORT` | `8000` | TCP Port to listen for HTTP requests on |
+| `UPSREAM_CRAN_SERVER_URL` | `https://cloud.r-project.org/` | URL of CRAN server we are proxying. <span class="underline">Include trailing slash</span> |
```
Comment (davidread) [inline README.md]: UPSREAM_CRAN_SERVER_URL should have a T in it I think

File: README.md [hunk 5]
```diff
@@ -1,4 +1,158 @@
...
+
+### Deployment<a id="sec-4-2-2"></a>
+
+This proxy can be deployed to a Kubernetes cluster using Helm using the
+corresponding
+[chart](https://github.com/ministryofjustice/analytics-platform-helm-charts/tree/master/charts/cran-proxy).
+
+# API<a id="sec-5"></a>
+
+-   The by default the proxy returns a HTTP 302 to the upstream
```
Comment (davidread) [inline README.md]: stray "The"

Comment (davidread) [inline README.md]: Maybe put the default last in the list? i.e. if-elif-else logic

**Review by davidread — APPROVED** [review]
All good, just some suggestions

File: README.md (hunk 1 above)
Comment (r4vi) [reply README.md]: they are.. my editor autogenerated it I'll link to the names instead

**Pull request merged**